orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -57,6 +58,7 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(T
       transformers.emplace_back(onnxruntime::make_unique<GeluFusion>(compatible_eps));
       transformers.emplace_back(onnxruntime::make_unique<LayerNormFusion>(compatible_eps));
       transformers.emplace_back(onnxruntime::make_unique<FastGeluFusion>(compatible_eps));
+      transformers.emplace_back(onnxruntime::make_unique<ConstantFolding>(compatible_eps, weights_to_train));
```

Comment:
>ConstantFolding [](start = 57, length = 15)

I think we used to hit a problem with constant folding when run with mixed precision mode. 

Constant folding is triggering CPU kernel, and we don't have fp16 implemented on CPU. 

I think you are not hitting this problem, coz this added to PreTrainingTransformer, which is trigger before MixedPrecisionTransformation.

Reply:
Yes. Checked the code again, the above UnsqueezeElimination should also be removed because of duplicated of this one.